mewarpx/scripts_ancillary/tinyprofiler_plotting_example.py

```diff
@@ -0,0 +1,270 @@
+
+As well as converting all the .out files in a directory to JSON files.
+'''
+
+import hatchet as ht
+import json
+import numpy as np
+import os
+import math
+from pathlib import Path
+import pandas as pd
+import matplotlib.pyplot as plt
+from matplotlib.pyplot import cm
+
+from mewarpx.utils_store import profileparser
+
+
+def get_parameter_name(exclusive, average):
+    parameter = ""
+    if exclusive:
+        parameter += "excl_"
+    else:
+        parameter += "incl_"
+
+    if average:
+        parameter += "avg"
+    else:
+        parameter += "max"
+
+    return parameter
+
+def get_description(exclusive, average):
+    description = ""
+    if exclusive:
+        description += "exclusive "
+    else:
+        description += "inclusive "
+
+    if average:
+        description += "average"
+    else:
+        description += "max"
+
+    return description
+
+def plot_max_percents(exclusive):
+    dataframes = []
+    parameter = "excl_max_percent" if exclusive else "incl_max_percent"
+    description = "Exclusive" if exclusive else "Inclusive"
+    for name in Path("warpx_outputs/").rglob("*.json"):
+        print("plotting:", name)
+
+        f = open(name)
+        data = json.load(f)
+
+        gf = ht.GraphFrame.from_literal(data)
+        gf.drop_index_levels()
+
+        split_name = str(name).split("/")
+        cells = split_name[1]
+        cores = split_name[2].replace("core", "")
+
+        # sort by excl_max_percent
+        gf.dataframe = gf.dataframe.sort_values(by=parameter, ascending=False)
+        # filter to only high, exclusive percents
+        filtered_gf = gf.filter(lambda x : not math.isnan(x[parameter]) and x[parameter] > 10)
+        # Add an 'info' index, which will be the pivot point later
+        filtered_gf.dataframe["info"] = f"{cells}x{cells}\n{cores}"
+
+        if "gpu" not in cores:
+            filtered_gf.dataframe["info"] += "procs"
+
+        # add the dataframe to the list
+        dataframes.append(filtered_gf.dataframe)
+
+    # concat all the dataframes together
+    result = pd.concat(dataframes)
+    # pivot the dataframes around the 'info' index
+    pivot_df = result.pivot(index="info", columns="name", values=parameter)
+
+    # plot the data
+    fig = pivot_df.loc[:,:].plot.bar(stacked=True, figsize=(11,12),
+        title=description + " max percents > 10%",
+        ylabel=description + " max percent", xlabel="Simulation Parameters",
+    )
+
+    # this puts the legend outside of the plot
+    fig.legend(bbox_to_anchor=(1,1))
+
+    plt.savefig(description.lower() + "_max_percent_time.png", bbox_inches="tight")
+
+def plot_union_of_times(top_n, exclusive, avg):
+    function_names = set([])
+    parameter = get_parameter_name(exclusive, avg)
+    description = get_description(exclusive, avg)
+
+    for name in Path("warpx_outputs/").rglob("*.json"):
+        f = open(name)
+        data = json.load(f)
```

Comment:
`with` block again - and consider abstracting this loading json / loading hatchet graph frame into a separate function.